src/main/asciidoc/_chapters/ops_mgt.adoc

```diff
@@ -3479,6 +3478,27 @@ To enable ACL, add the following to your hbase-site.xml and restart your Master:
 <property>
 ----
 
+=== Migrating From Old Implementation
+The coprocessor `org.apache.hadoop.hbase.rsgroup.RSGroupAdminEndpoint` is
+deprected, but for compatible, if you want the pre 3.0.0 hbase client/shell
+to communicate with the new hbase cluster, you still need to add this
+coprocessor to master. And if this coprocessor is specified, the
+`hbase.balancer.rsgroup.enabled` flag will be set to true automatically to
```

Comment:
if they're relying on this side effect of configuring the coprocessor, do they also have to be sure to set the custom balancer (as in the old implementation) or does the flag flipping to true take care of that?

Reply:
No they don't. Actually I think there are two requirements. One is for communicating with pre 3.0.0 hbase client/shell, then you have to set the coprocessor. Another is that you upgrade from pre 3.0.0 to 3.0.0, then we will set the flag to true automatically for you if you have already enabled rs group by the old way.

Reply:
okay good. Presuming there's no harm in still having the balancer customized I think this is fine. If having the balancer customized still is a problem, then we should include a sentence that says "you should remove the custom balancer config"

Reply:
Let me confirm. Now the top level balancer will always be RSGroupBasedBalancer, and this option is for the balancer used within a rsgroup. So typically it is not allowed to configured RSGroupBasedBalancer again. But I think a warning here is enough.
And maybe we should also add something in the balancer section to mention that RSGroupBasedBalancer is now the only top level balancer?

Reply:
OK, we have this in RSGroupBasedLoadBalancer.initialize

```
    // Create the balancer
    Class<? extends LoadBalancer> balancerClass;
    String balancerClassName = config.get(HBASE_RSGROUP_LOADBALANCER_CLASS);
    if (balancerClassName == null) {
      balancerClass = config.getClass(HConstants.HBASE_MASTER_LOADBALANCER_CLASS,
        LoadBalancerFactory.getDefaultLoadBalancerClass(), LoadBalancer.class);
    } else {
      try {
        balancerClass = Class.forName(balancerClassName).asSubclass(LoadBalancer.class);
      } catch (ClassNotFoundException e) {
        throw new IOException(e);
      }
    }
    // avoid infinite nesting
    if (getClass().isAssignableFrom(balancerClass)) {
      balancerClass = LoadBalancerFactory.getDefaultLoadBalancerClass();
    }
    internalBalancer = ReflectionUtils.newInstance(balancerClass);
```

Reply:
Checked the ref guide, we haven't mentioned much about the balancer class I think making the top level balancer to be RSGroupBasedLoadBalancer can be considered as an 'implementation detail' so do not mention it in ref guide is fine I think.

Reply:
I would agree in general, but the instructions for using the regroup feature expressly called out configuring the balancer. So in this paragraph about migrating we need to call out that they should no longer be doing this.

Reply:
Done.